lib/search-suggestions/index.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import SmallSearchIcon from '../icons/search-small';
+import TagIcon from '../icons/tag';
+
+export class SearchSuggestions extends Component {
+  static displayName = 'SearchSuggestions';
+
+  static propTypes = {
+    onSearch: PropTypes.func.isRequired,
+    query: PropTypes.string.isRequired,
+    filteredTags: PropTypes.array.isRequired,
+    storeKeyHandler: PropTypes.func.isRequired,
+  };
+
+  state = {
+    selectedItem: 0,
+  };
+
+  componentDidMount() {
+    this.props.storeKeyHandler({
+      next: this.nextItem,
+      prev: this.prevItem,
+      select: this.selectItem,
+    });
+  }
+
+  componentDidUpdate() {
+    this.props.storeKeyHandler({
+      next: this.nextItem,
+      prev: this.prevItem,
+      select: this.selectItem,
+    });
+  }
+
+  nextItem = () => {
+    const { filteredTags } = this.props;
+    const { selectedItem } = this.state;
+    const newItem =
+      selectedItem === filteredTags.length
+        ? filteredTags.length
+        : selectedItem + 1;
```

Comment:
I'm not sure what we're trying to do here. It looks like this will lead us to pick an index just past the end of the array. If we're trying to _clamp_ the index to the length of the array we can do so in one step with a little math

```js
const newItem = Math.min( selectedItem + 1, filteredTags.length - 1 );
```

Reply:
Oh it's because the search is the first select-able element... then the tag array... so it's the index plus one. This could be clearer I admit >_>